src/vs/workbench/contrib/terminal/browser/terminalEditorService.ts

```diff
@@ -158,18 +158,19 @@ export class TerminalEditorService extends Disposable implements ITerminalEditor
 		this._onDidChangeActiveInstance.fire(this.activeInstance);
 	}
 
-	async openEditor(instance: ITerminalInstance, sideGroup: boolean = false): Promise<void> {
+	async openEditor(instance: ITerminalInstance, sideGroup: boolean = false, editorOptions?: TerminalEditorLocation): Promise<void> {
 		const resource = this.resolveResource(instance);
 		if (resource) {
 			await this._editorService.openEditor({
 				resource,
 				options:
 				{
 					pinned: true,
-					forceReload: true
+					forceReload: true,
+					preserveFocus: editorOptions?.preserveFocus
 				}
 			},
-				sideGroup ? SIDE_GROUP : undefined);
+				editorOptions?.viewColumn || sideGroup ? SIDE_GROUP : undefined);
```

Comment:
Let's move this up one line and add parens around `(sideGroup ? SIDE_GROUP : undefined)` for clarify